java/s2i/assemble

```diff
@@ -93,17 +114,25 @@ function build_maven() {
   mvn ${maven_env_args} ${maven_args} ${MAVEN_ARGS_APPEND}
   check_error "Maven build" $?
 
-  # ======================
-  # After-build validation
-  if [ ! -d "${build_dir}/hawt-app" ] && [ x"${JAVA_MAIN_CLASS}" = x ]; then
-    echo "No JAVA_MAIN_CLASS specified and no hawt-app-maven-plugin configured in the pom.xml"
-    exit 1
-  fi
-
   # ==============
   # Copy artifacts
-  echo "Copying Maven artifacts from ${build_dir}/hawt-app to ${app_dir} ..."
-  copy_dir "${build_dir}/hawt-app" ${app_dir}
+  if [ -d "${build_dir}/hawt-app" ]; then
+    echo "Hawtapp plugin detected. Copying Maven artifacts from ${build_dir}/hawt-app to ${app_dir} ..."
```

Comment:
This works only when hawt-app is configured in the pom.xml, right ? The previous version worked also with hawt-app when only a JAVA_MAIN_CLASS is given and no configuration of hawt-app within the pom.xml (not that I say that was a good solution, just saying /wrt to backwards compatibility ;-)